### ISSUE TITLE
Version extraction code is incorrect

### ISSUE THREAD

**Issue by serge-sans-paille** [man-group/sparrow#167]
cmake log returns

```
-- Building sparrow v..

```

A possible fix vould be:

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 419fbe3..83f51d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,7 +33,7 @@ set(SPARROW_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
 file(STRINGS "${SPARROW_INCLUDE_DIR}/sparrow/config/sparrow_version.hpp" sparrow_version_defines
      REGEX "constexpr int SPARROW_VERSION_(MAJOR|MINOR|PATCH)")
 foreach(ver ${sparrow_version_defines})
-    if(ver MATCHES "constexpr int SPARROW_VERSION_(MAJOR|MINOR|PATCH) +([^ ]+)$")
+    if(ver MATCHES "constexpr int SPARROW_VERSION_(MAJOR|MINOR|PATCH) = ([0-9]+);$")
         set(SPARROW_VERSION_${CMAKE_MATCH_1} "${CMAKE_MATCH_2}" CACHE INTERNAL "")
     endif()
 endforeach()
```

**Comment by SylvainCorlay** [comment]
I am surprised that we don't see this issue in other packages that use the same regex.

**Comment by serge-sans-paille** [comment]
it's visible on CI too: https://github.com/man-group/sparrow/pull/165/checks#step:7:15

**Comment by SylvainCorlay** [comment]
Fixed in #168.
